libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarterOptions.kt

```diff
@@ -0,0 +1,93 @@
+package com.mapbox.navigation.core.trip
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.replay.route.ReplayRouteSessionOptions
+
+/**
+ * Defines options for the automatic trip starter [MapboxTripStarter].
+ *
+ * @param tripType Specify the type of trip to start.
+ * @param replayRouteSessionOptions options to use when isReplayRouteEnabled is true
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class MapboxTripStarterOptions private constructor(
+    @MapboxTripStarterExtra.Type
+    val tripType: String,
+    val replayRouteSessionOptions: ReplayRouteSessionOptions,
```

Comment:
I don't think these two properties fit together in one class. I would add functions to `MapboxTripStarter` like `setTypeType` and `getTripType` and move `tripType` property to `MapboxTripStarterState`. `MapboxTripStarterOptions` can be removed in favor of using `ReplayRouteSessionOptions` directly.

Reply:
I was considering the "functions instead of options" approach, I'll see if that simplifies the internal logic. Options are consistent with many other features so I have preferred them. I actually want to delete `MapboxTripStarterState` in favor of a better `LocationPermissionsProvider`, the only reason it exists is because location permissions are not observable (only sync get functions)

Related to this conversation https://github.com/mapbox/mapbox-navigation-android/pull/6794/files#r1065486611. There is an issue with the internal logic, so maybe will change all this in order to make it work.

Basically, what should happen when you change the ReplayRouteSessionOptions in the middle of a route simulation? Does it restart? Does it update the simulation?.. Dynamically updating simulation is not supported yet so I think it would have to restart.

Reply:
I think it would be better to go with functions. `isLocationPermissionGranted` and `tripType` look rather related to me, because they both control the trip session state (stopped, started or started with replay), but they are set differently now. `tripType` and `replayRouteSessionOptions` are not that related in my opinion. Updating `tripType` is a more common scenario than updating `replayRouteSessionOptions`, but you can only update them simultaneously now. You mitigated that by adding `update` extension, but that wouldn't be necessary with separate functions. Restarting the simulation if `replayRouteSessionOptions` is fine to me (as I see it doesn't happen at the moment) and that likely would be easier to implement with functions (no need to check which part of `MapboxTripStarterOptions` changed).

Reply:
Ok i'll go with the functions instead of options approach. Options have advantages, but in this case it seems to be overkill. We always have the ability to add them if we need them, but will start with just functions and internal state.

Reply:
> I actually want to delete MapboxTripStarterState in favor of a better LocationPermissionsProvider, the only reason it exists is because location permissions are not observable (only sync get functions)

Instead of the `MapboxTripStarterState` can we not directly store `isLocationPermissionGranted` in `MapboxTripStarter`? I agree `MapboxTripStarterState` looks redundant  and it would be good to get rid of it. @kmadsen

Reply:
Yeah I liked how it kept all of the state for this class in one place. But it isn't really needed. Removed the object.